File: src/main/java/net/andrecarbajal/naviMusic/util/URLUtils.java
package net.andrecarbajal.naviMusic.util;

import lombok.experimental.UtilityClass;

import java.net.URI;
import java.util.Optional;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@UtilityClass
public class URLUtils {
    public Optional<String> getURLParam(String url, String param) {
        final String regex = param + "=([^&]*)";

        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(url);

        return matcher.find() ? Optional.of(matcher.group(1)) : Optional.empty();
    }

    public boolean isURL(String input) {
        try {
            new URI(input);
            return true;
        } catch (Exception ignored) {
            return false;
        }
    }
}

File: src/main/kotlin/net/andrecarbajal/naviMusic/util/URLUtils.kt
package net.andrecarbajal.naviMusic.util

import java.net.URI
import java.util.*
import java.util.regex.Pattern

object URLUtils {
    @JvmStatic
    fun getURLParam(url: String, param: String): Optional<String> {
        val regex = "$param=([^&]*)"
        val pattern = Pattern.compile(regex)
        val matcher = pattern.matcher(url)

        return if (matcher.find()) Optional.of(matcher.group(1)) else Optional.empty()
    }

    @JvmStatic
    fun isURL(input: String?): Boolean {
        if (input == null) return false
        return try {
            URI(input)
            true
        } catch (_: Exception) {
            false
        }
    }
}
